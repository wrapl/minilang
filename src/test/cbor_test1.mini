let S := string::buffer()

S:write("fun(X) X + {\n\t\"A1\" is 1")
for I in 2 .. 100 do
	S:write(',\n\t\"A{I}\" is {I}')
end
S:write("\n}")

let C := compiler(globals())
let P := parser()
P:source("test", 1)
P:input(S:rest)

let F1 := P:compile(C)()

let E := fmt::cbor::encode(F1)

print(enc::base16::encode(E), "\n")

let F2 := fmt::cbor::decode(E)

print(F2({}))
