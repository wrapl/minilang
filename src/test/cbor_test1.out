D81B82612AD81B986861215901580A0E40746573743A31087465737402CE02000002580202580102000000F1B6F1B4003A012A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A02322A020AC8527604600102030405060708090A0B0C0D0E0F101112131415161718181819181A181B181C181D181E181F1820182118221823182418251826182718281829182A182B182C182D182E182F1830183118321833183418351836183718381839183A183B183C183D183E183F1840184118421843184418451846184718481849184A184B184C184D184E184F1850185118521853185418551856185718581859185A185B185C185D185E185F18601861186218631864D81B98656D6D61703A3A74656D706C6174656241316241326241336241346241356241366241376241386241396341313063413131634131326341313363413134634131356341313663413137634131386341313963413230634132316341323263413233634132346341323563413236634132376341323863413239634133306341333163413332634133336341333463413335634133366341333763413338634133396341343063413431634134326341343363413434634134356341343663413437634134386341343963413530634135316341353263413533634135346341353563413536634135376341353863413539634136306341363163413632634136336341363463413635634136366341363763413638634136396341373063413731634137326341373363413734634137356341373663413737634137386341373963413830634138316341383263413833634138346341383563413836634138376341383863413839634139306341393163413932634139336341393463413935634139366341393763413938634139396441313030D827612B
{A1 is 1, A2 is 2, A3 is 3, A4 is 4, A5 is 5, A6 is 6, A7 is 7, A8 is 8, A9 is 9, A10 is 10, A11 is 11, A12 is 12, A13 is 13, A14 is 14, A15 is 15, A16 is 16, A17 is 17, A18 is 18, A19 is 19, A20 is 20, A21 is 21, A22 is 22, A23 is 23, A24 is 24, A25 is 25, A26 is 26, A27 is 27, A28 is 28, A29 is 29, A30 is 30, A31 is 31, A32 is 32, A33 is 33, A34 is 34, A35 is 35, A36 is 36, A37 is 37, A38 is 38, A39 is 39, A40 is 40, A41 is 41, A42 is 42, A43 is 43, A44 is 44, A45 is 45, A46 is 46, A47 is 47, A48 is 48, A49 is 49, A50 is 50, A51 is 51, A52 is 52, A53 is 53, A54 is 54, A55 is 55, A56 is 56, A57 is 57, A58 is 58, A59 is 59, A60 is 60, A61 is 61, A62 is 62, A63 is 63, A64 is 64, A65 is 65, A66 is 66, A67 is 67, A68 is 68, A69 is 69, A70 is 70, A71 is 71, A72 is 72, A73 is 73, A74 is 74, A75 is 75, A76 is 76, A77 is 77, A78 is 78, A79 is 79, A80 is 80, A81 is 81, A82 is 82, A83 is 83, A84 is 84, A85 is 85, A86 is 86, A87 is 87, A88 is 88, A89 is 89, A90 is 90, A91 is 91, A92 is 92, A93 is 93, A94 is 94, A95 is 95, A96 is 96, A97 is 97, A98 is 98, A99 is 99, A100 is 100}